cmd/juju/gui/gui.go

```diff
@@ -151,6 +154,20 @@ func (c *guiCommand) Run(ctx *cmd.Context) error {
 	return nil
 }
 
+// guiAddr returns an address where the GUI is available.
+func guiAddr(ctx *cmd.Context, conn api.Connection) string {
+	client := controller.NewClient(conn)
+	config, err := client.ControllerConfig()
+	if err != nil {
+		ctx.Warningf("cannot get controller config: %s", err)
+		return conn.Addr()
+	}
+	if dnsName := config.AutocertDNSName(); dnsName != "" {
```

Comment:
I think we have have this info already available in conn.PublicDNSName without making an API call that might fail.

Reply:
Done.